Documentation/ApiOverview/ContentElements/CustomDataProcessing.rst

```diff
@@ -0,0 +1,157 @@
+   all configuration of :typoscript:`tt_content.examples_dataproccustom`
+
+:php:`array $processorConfiguration`
+   Contains the configuration of the currently called dataprocessor. In this
+   case the value of :typoscript:`as` and the stdWrap configuration of the
+   :typoscript:`categoryList`
+
+
+:php:`array $processedData`
+   On calling containt the processed data of all previously called data 
+   processors on this content element. Your custom data processor also stores
+   the variables to be send to Fluid here.
+
+This is an example implementation of a custom data processor::
+
+   namespace T3docs\Examples\DataProcessing;
+
+   use TYPO3\CMS\Core\Utility\GeneralUtility;
+   use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+   use TYPO3\CMS\Frontend\ContentObject\DataProcessorInterface;
+   use TYPO3\CMS\Extbase\Domain\Repository\CategoryRepository;
+
+   class CustomCategoryProcessor implements DataProcessorInterface
+   {
+      public function process : array (
+         ContentObjectRenderer $cObj,
+         array $contentObjectConfiguration,
+         array $processorConfiguration,
+         array $processedData
+      ) {
+         if (isset($processorConfiguration['if.']) && !$cObj->checkIf($processorConfiguration['if.'])) {
+            // leave $processedData unchanged in case there were previous other processors
+            return $processedData;
+         }
+         // categories by comma separated list
+         $categoryIdList = $cObj->stdWrapValue('categoryList', $processorConfiguration ?? []);
+         if ($categoryIdList) {
+            $categoryIdList = GeneralUtility::intExplode(',', (string)$categoryIdList, true);
+         }
+
+         /** @var CategoryRepository $categoryRepository */
+         $categoryRepository = GeneralUtility::makeInstance(CategoryRepository::class);
+         $categories = [];
+         foreach ($categoryIdList as $categoryId) {
+            $categories[] = $categoryRepository->findByUid($categoryId);
+         }
+         // set the categories into a variable, default "categories"
+         $targetVariableName = $cObj->stdWrapValue('as', $processorConfiguration, 'categories');
+         $processedData[$targetVariableName] = $categories;
+         return $processedData;
+      }
+   }
+
+In the extension *examples* you can finde the code in
+:file:`typo3conf/ext/examples/Classes/DataProcessing/CustomCategoryProcessor.php`.
+
+On being called the :php:`CustomCategoryProcessor` runs stdWrap on the calling
+ContentObjectRenderer, which has the data of the table :php:`tt_content` the
+calling content element.
+
+Since the field :php:`categoryList` got configured in TypoScript as follows:
+
+..code-block:: typoscript
+
+   categoryList.field = tx_examples_main_category
+
+:ref:`stdWrap <t3tsref:stdwrap>` fetches the value of :php:`categoryList` from
+:php:`tt_content.tx_examples_main_category` of the currently calling content
```

Comment:
```suggestion
:php:`tt_content.tx_examples_main_category` of the currently called content
```

not sure?

Reply:
actually the CE is active here, the caller, not the one to be called. So it would be the calling content element?